File: integration-test-java/src/main/java/org/seasar/doma/it/criteria/Address.java
package org.seasar.doma.it.criteria;

import org.seasar.doma.Entity;
import org.seasar.doma.Id;
import org.seasar.doma.Metamodel;

@Entity(metamodel = @Metamodel)
public class Address {
  @Id private Integer addressId;
  private Street street;
  private Integer version;

  public Integer getAddressId() {
    return addressId;
  }

  public void setAddressId(Integer addressId) {
    this.addressId = addressId;
  }

  public Street getStreet() {
    return street;
  }

  public void setStreet(Street street) {
    this.street = street;
  }

  public Integer getVersion() {
    return version;
  }

  public void setVersion(Integer version) {
    this.version = version;
  }
}

File: integration-test-kotlin/src/main/kotlin/org/seasar/doma/it/criteria/Address.kt
package org.seasar.doma.it.criteria

import org.seasar.doma.Entity
import org.seasar.doma.Id
import org.seasar.doma.Metamodel

@Entity(metamodel = Metamodel())
class Address {
    @Id
    var addressId: Int? = null
    var street: Street? = null
    var version: Int? = null
}
